apps/law_memory/metrics.py

```
@@ -0,0 +1,26 @@
+from __future__ import annotations
+
+from typing import Dict
+
+import numpy as np
+
+
+def inject_into_metrics(r: np.ndarray, law: np.ndarray, xi: float = 0.382, lambda_h: float = 0.1457) -> Dict[str, float]:
+    """Return updated metric coefficients from a law profile."""
+
+    if r.size != law.size:
+        raise ValueError("grid mismatch")
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)